src/runtime/runtime.js

```diff
@@ -56,14 +57,14 @@
   // This has to come before any function below that uses tail recursion.
   // Every call is the following functions is deliberately not in tail position.
 
-  var counter = 0;
+  var counter = Date.now() % 1e9;
 
   /**
    * Generates a new unique string.
    * @return {string}
    */
   function newUniqueString() {
-    return '__$' + Math.floor(Math.random() * 1e9) + '$' + ++counter + '$__';
+    return '__$' + (random() * 1e9 >>> 1) + '$' + ++counter + '$__';
```

Comment:
A reference to the analysis for this construct or an analysis in a comment would help clarify the design.